### PR TITLE
Deployment count cap

### PR DESCRIPTION
**Pull request by aalavandhan**
* Added deployment count limit to prevent large loops which might run out of gas

### CODE REVIEW

**Review by brandoniles — COMMENTED** [review]
I see the problem of unbounded loops causing gas limit problems, but I'm not sure I see this as a right solution for it. If there's no hard limit in the code, then it will be implicitly enforced anyway by the block gas limit. This change would simply make the limit lower?

I could see this working if, instead of reverting, it executed as much as it could up to some limit, then left the rest for followup calls, though.

I'm guessing if either this code limit or block gas limit were reached, it could brick the contract until somehow the tranches were redeemed from the side. To make sure we have a recovery step--Is there a non-permissioned way to redeem the tranches to underlying for the vault?

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 1]
```diff
@@ -446,6 +446,9 @@ contract RolloverVault is
         if (balance > 0 && !isHeld) {
             // Inserts new token into the deployed assets list.
             _deployed.add(address(token));
+            if (_deployed.length() > MAX_DEPLOYED_COUNT) {
```
Comment (aalavandhan) [inline spot-contracts/contracts/vaults/RolloverVault.sol:449]: This is on deployment

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 2]
```diff
@@ -75,6 +72,9 @@ contract RolloverVault is
     uint8 private constant PERP_PRICE_DECIMALS = 8;
     uint256 private constant PERP_UNIT_PRICE = (10**PERP_PRICE_DECIMALS);
 
+    /// @dev The maximum number of deployed assets that can be held in this vault at any given time.
+    uint256 public constant MAX_DEPLOYED_COUNT = 47;
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:76]: It was just a random odd number, but safe enough to not be too high or low. 

Redeeming 47 assets costs around 2m in gas. I didn't want too large of a number which is close to the block gas limit (which could change up or down). I don't see any scenario where we have over 10 assets in the vault.

File: spot-contracts/contracts/vaults/RolloverVault.sol (hunk 1 above)
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:449]: Yea exactly

File: spot-contracts/contracts/vaults/RolloverVault.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (brandoniles) [reply spot-contracts/contracts/vaults/RolloverVault.sol:76]: 2m gas out of 15m seems safe enough

**Review by brandoniles — APPROVED** [review]
LGTM

**Pull request merged**